Index: src/ui.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/ui.cpp b/src/ui.cpp
--- a/src/ui.cpp	
+++ b/src/ui.cpp	(date 1727707071940)
@@ -38,7 +38,8 @@
 }
 
 void UI::add_menu(Menu menu_) {
-    m_menu[menu_.]
+    m_menu[menu_.name()][menu_];
+    m_menu.insert(menu_);
 }
 
 Menu::Menu(std::string name_, std::vector<std::string> list_) : m_name(name_), m_list(list_){
@@ -50,5 +51,5 @@
 }
 
 std::string Menu::name() {
-    return std::string();
+    return m_name;
 }
